libctru/include/3ds/services/pmapp.h

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file pmapp.h
+ * @brief PM (Process Manager) application service.
+ */
+#pragma once
+
+#include <3ds/services/fs.h>
+#include <3ds/exheader.h>
+
+/// Launch flags for PM launch commands.
+enum {
+	PMLAUNCHFLAG_NORMAL_APPLICATION            = BIT(0),
+	PMLAUNCHFLAG_LOAD_DEPENDENCIES             = BIT(1),
+	PMLAUNCHFLAG_NOTIFY_TERMINATION            = BIT(2),
+	PMLAUNCHFLAG_QUEUE_DEBUG_APPLICATION       = BIT(3),
+	PMLAUNCHFLAG_TERMINATION_NOTIFICATION_MASK = 0xF0,
+	PMLAUNCHFLAG_FORCE_USE_O3DS_APP_MEM        = BIT(8),  ///< Forces the usage of the O3DS system mode app memory setting even if N3DS system mode is not "Legacy". Dev4 and Dev5 not supported. N3DS only.
+	PMLAUNCHFLAG_FORCE_USE_O3DS_MAX_APP_MEM    = BIT(9),  ///< In conjunction with the above, forces the 96MB app memory setting. N3DS only.
+	PMLAUNCHFLAG_USE_UPDATE_TITLE              = BIT(16),
+};
+
+/// Initializes pm:app.
+Result pmAppInit(void);
+
+/// Exits pm:app.
+void pmAppExit(void);
+
+/**
+ * @brief Gets the current pm:app session handle.
+ * @return The current pm:app session handle.
+ */
+Handle *pmAppGetSessionHandle(void);
+
+/**
+ * @brief Launches a title.
+ * @param programInfo Program information of the title.
+ * @param launchFlags Flags to launch the title with.
+ */
+Result PMAPP_LaunchTitle(const FS_ProgramInfo *programInfo, u32 launchFlags);
+
+/**
+ * @brief Launches a title, applying patches.
+ * @param programInfo Program information of the title.
+ * @param programInfoUpdate Program information of the update title.
+ * @param launchFlags Flags to launch the title with.
+ */
+Result PMAPP_LaunchTitleUpdate(const FS_ProgramInfo *programInfo, const FS_ProgramInfo *programInfoUpdate, u32 launchFlags);
+
+/**
+ * @brief Gets a title's ExHeader Arm11CoreInfo and SystemInfo flags.
+ * @param programInfo Program information of the title.
+ * @param[out] outCoreInfo Pointer to write the ExHeader Arm11CoreInfo to.
+ * @param[out] outSiFlags Pointer to write the ExHeader SystemInfo flags to.
+ */
+Result PMAPP_GetTitleExheaderFlags(ExHeader_Arm11CoreInfo* outCoreInfo, ExHeader_SystemInfoFlags* outSiFlags, const FS_ProgramInfo *programInfo);
```

Comment:
`programInfo` should probably the first param like implied in the doc comment; *or* the doc should be updated.